token-metadata/program/src/utils.rs

```diff
@@ -550,26 +550,35 @@ pub fn calculate_supply_change<'a>(
     // instead of properly tracking the supply.
     // Now, we increment this by one if the edition number is less than the max supply.
     // This allows users to mint out missing edition numbers that are less than the supply, but
-    // tracks the supply correctly once all missing editions are minted.
+    // tracks the supply correctly for new Master Editions.
     let new_supply = if let Some(max_supply) = max_supply {
+        // We should never be able to mint an edition number that is greater than the max supply.
+        if edition > max_supply {
```

Comment:
So I guess this was needed because the instructions to mint print editions look like they would have just allocated a new editionMarker account if the edition was out of bounds from existing ones?  So no other checks would have caught that.